std/experimental/allocator/package.d

```diff
@@ -760,19 +764,23 @@ if (isInputRange!R)
         auto result = cast(T[]) m;
 
         size_t i = 0;
-        scope (failure)
+
+        //scope (failure) alloc.deallocate(m);
```

Comment:
if we really can't set deallocate with a GC to `@trusted`, then allocating with a forward range can't be safe either :/
Any ideas?